pycc/cceom.py

```diff
@@ -0,0 +1,315 @@
+
+            # Compute sigma vectors
+            s1 = np.zeros_like(C1)
+            s2 = np.zeros_like(C2)
+            for state in range(nvecs):
+                s1[state] = self.s1(hbar, C1[state], C2[state])
+                s2[state] = self.s2(hbar, C1[state], C2[state])
+            sigma_done = M
+
+            # Build and diagonalize subspace Hamiltonian
+            S = np.vstack((S, np.hstack((np.reshape(s1, (nvecs, no*nv)), np.reshape(s2, (nvecs, no*no*nv*nv))))))
+            G = C @ S.T
+            E, a = np.linalg.eig(G)
+
+            # Sort eigenvalues and corresponding eigenvectors into ascending order
+            idx = E.argsort()[:N]
+            E = E[idx]; a = a[:,idx]
+
+            # Build correction vectors
+            r = a.T @ S - np.diag(E) @ a.T @ C
+            r_norm = np.linalg.norm(r, axis=1)
+            delta = r/np.subtract.outer(E,D) # element-by-element division
+
+            # Print status and check convergence and print status
+            dE = E - E_old
+            print("             E/state                   dE                 norm")
+            for state in range(N):
+                print("%20.12f %20.12f %20.12f" % (E[state], dE[state], r_norm[state]))
+
+            if (np.abs(np.linalg.norm(dE)) <= e_conv):
+                converged = True
+                break
+
+            if M >= maxM:
+                # Collapse to N vectors if subspace is too large
+                print("\nMaximum allowed subspace dimension (%d) reached. Collapsing to N roots." % (maxM))
+                C = a.T @ C
+                M = N
+                E = E_old
+                sigma_done = 0
+                S = np.empty((0,sigma_len), float)      
+            else:
+                # Add new vectors to guess space
+                C = np.concatenate((C, delta[:N]))
+
+        if converged:
+            print("\nCCEOM converged in %.3f seconds." % (time.time() - time_init))
+            print("\nState     E_h           eV")
+            print("-----  ------------  ------------")
+            eVconv = psi4.qcel.constants.get("hartree energy in ev")
+            for state in range(N):
+                print("  %3d  %12.10f  %12.10f" %(state, E[state], E[state]*eVconv))
+
+            return E, C
+
+
+    def guess(self, M, o, v, method):
+        """
+        Compute guess vectors for EOM-CC Davidson algorithm
+
+        Parameters
+        ----------
+        M : int
+            number of guesses to generate
+        o, v : NumPy slices
+            slices for occupied and virtual spaces, respectively
+        method : str
+            choice of method to generate guesses
+
+        Returns
+        -------
+        eps : NumPy array
+            eigenvalues/energies associated with guess vectors
+        guesses : NumPy array
+            guess vectors (as rows of matrix)
+        """
+        no = o.stop - o.start
+        nv = v.stop - v.start
```

Comment:
I believe here you should be pulling `no` and `nv` from the `ccwfn` object (or `cchbar`, since they are copied there as well), which are the numbers of _active_ occupied / virtual orbitals. without this, `freeze_core = True` will throw `ValueError: operands could not be broadcast together`.

Reply:
Yeah, I originally wrote the `guess` function to be independent of the `cceom` class, but then thought better of it when I decided to make the guesses a little more sophisticated.  Everything is pulled from the `cchbar` object and frozen core works correctly (according to the test case).